Workaround `woocommerce/data` dependency issues, reset `package-lock.json`

tomalec opened the pull request:
### Changes proposed in this Pull Request:

- Make webpack resolve `[tj]sx` files as well, (1c795ae)
	to workaround `@woocommerce/data` not building after `package-lock.json` reset.

- Add `md5` as dependency, (af2c472)
	as `@woocommerce/data` needs it and does not state it.
	Workaround woocommerce/woocommerce-admin#7086
- Reset `package-lock.json` (06904da)

- Add missing dependencies to `package.json`, (0a09423)
- Add `eslint-import-resolver-webpack` dep to resolve our `.~/` alias for lining (ff4cecd)
- Make `no-unsafe-wp-apis` eslint rule warn iInstead of throwing an error. (236e1c1)
- Add more packages to jest + wc-admin workaround. (0394c06)
	Add `experimental` and all combinations of `package1/node_modules/@woocommerce/package2`.
- Add `@wordpress/core-data` to dependencies. (d4833e4)
	As it is required peer for `@woocommerce/data@1.3.0`.
- Update `@woocommerce/eslint-plugin@1.2.0` dev dependency (1da555c)
	from `^1.0.0-beta.0`, as we already match its requirements.



### Screenshots:

n/a


### Detailed test instructions:

1. Checkout this branch
2. Remove `package-lock.json`
3. install and build
	  ```
	  npm install
	  npm run start
	  ```
4. You should see no build errors.

### Changelog Note:

> Fix dependencies for a clean build after `package-lock.json` reset.

tomalec committed:
Make webpack resolve `[tj]sx` files  as well,

to workaround `@woocommerce/data` not building after `package-lock.json` reset.

tomalec committed:
Add `md5` as dependency,

as `@woocommerce/data` needs it and does not state it.
Work around https://github.com/woocommerce/woocommerce-admin/issues/7086

tomalec committed:
Reset `package-lock.json`

eason9487 reviewed (approved):
Tested well by removing `package-lock.json` only, and also both `package-lock.json` and `node_modules`. LGTM.

tomalec committed:
Merge branch 'trunk' into fix/woo-data-dependency

tomalec committed:
Add missing dependencies to package.json,

Add `components/node_modules/@woocommerce/currency` to transform list to `jest.config.js`

tomalec committed:
Add `eslint-import-resolver-webpack` dep to resolve our alias for linting.

tomalec committed:
Make `no-unsafe-wp-apis` eslint rule warn,

Instead of throwing an error.

tomalec committed:
Add more packages to jest + wc-admin workaround.

Add `experimental` and all combinations of `package1/node_modules/@woocommerce/package2`.

tomalec committed:
Add `@wordpress/core-data` to dependencies.

As it is required peer for `@woocommerce/data@1.3.0`.

Suppress the install warning and make wc-admin dependency tests pass.

tomalec committed:
Update `@woocommerce/eslint-plugin@1.2.0` dev dependency

from `^1.0.0-beta.0`

tomalec committed:
Lock dependencies on specific versions,

@woocommerce/data@>=1.1.1 <1.2.0 @wordpress/icons@>=2.9.1 <3.0.0@wordpress/primitives@>=1.11.1 <2.0.0
To prevent breaking changes and incompatibilities between deps result in bugs in runtime. (Giant icons, `TypeError` )

As suggested in https://github.com/woocommerce/google-listings-and-ads/pull/693#issuecomment-852029874

Co-authored-by: Eason <eason.su.tw@gmail.com>

eason9487 reviewed (approved):
Tested locally and works well. LGTM.
